routes/auth.routes.js

```diff
@@ -0,0 +1,83 @@
+const router = require('express').Router();
+const bcrypt = require('bcryptjs');
+const mongoose = require('mongoose');
+const User = require('../models/User.model');
+const isLoggedOut = require('../middleware/route-guard');
+
+router.get('/signup', (req, res) => res.render('auth/signup'));
+
+router.post('/signup', async (req, res, next) => {
+ try{
+    let {username, password} = req.body; 
+    if(!username || !password) {
+        res.render('auth/signup', {errorMessage: 'Please input all the fields'});
+    }
+    const salt = await bcrypt.genSalt(10);
+    const hashedPassword = await bcrypt.hash(password, salt);
+    await User.create({username, password: hashedPassword});
+    res.redirect('/');
+ } catch {
+    if(error instanceof mongoose.Error.ValidationError) {
+        res.render('auth/signup', {
+            errorMessage: error.message,
+        });
+    } else if (error.code === 11000) {
+        res.render('auth/signup', {
+            errorMessage: 'Email already registered',
+        });
+    }
+    
+    console.log(error);
+    next(error);
+ }
+});
+
+router.get('/login', (req, res) => res.render('auth/login'));
+
+router.post('/login', async (req, res , next) => {
+    try {
+        let {username, password} = req.body;
+        if (!username || !password) {
+            res.render('auth/login', {errorMessage: 'Please input all the fields'});
+        }
+        let user = await User.findOne({email});
```

Comment:
You're passing the email here, but when I signed up and logged into your website, there was no option to fill in my email. So the users you're creating are being created without the email. When we get to this route here, it will look for the email, and it will break because the users don't have it. Be more careful with that next time!